java_src/foam/core/AbstractFObject.java

```diff
@@ -7,6 +7,23 @@
   extends ContextAwareSupport
   implements FObject
 {
+  public FObject fclone() {
+    FObject ret;
+    try {
+      ret = (FObject) getClassInfo().getObjClass().newInstance();
+    } catch (InstantiationException | IllegalAccessException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
+    for(PropertyInfo pi : props) {
```

Comment:
See FOAM style guideline: https://github.com/foam-framework/foam/wiki/FOAM2-Style-Guidelines

This line should be:

for ( PropertyInfo pi : props ) {